webhookValidator.go

```diff
@@ -70,145 +70,75 @@ func (vf ValidatorFunc) Validate(w Webhook) error {
 	}
 }
 
-// GoodFailureURL parses the given webhook's FailureURL
-// and returns as soon as the URL is considered invalid. It returns nil if the URL is
-// valid.
-func GoodFailureURL(vs []ValidURLFunc) ValidatorFunc {
-	vs = filterNil(vs)
+// CheckDeviceID ensures that the DeviceIDs are able to parse into regex.
+func CheckDeviceID() ValidatorFunc {
 	return func(w Webhook) error {
-		if w.FailureURL != "" {
-			parsedFailureURL, err := url.ParseRequestURI(w.FailureURL)
+		for _, i := range w.Matcher.DeviceID {
+			_, err := regexp.Compile(i)
 			if err != nil {
-				return fmt.Errorf("%w: %v", errInvalidFailureURL, err)
-			}
-			for _, f := range vs {
-				err = f(parsedFailureURL)
-				if err != nil {
-					return fmt.Errorf("%w: %v", errInvalidFailureURL, err)
-				}
+				return errDeviceIDUnparseable
 			}
 		}
 		return nil
 	}
 }
 
-// GoodAlternativeURLs parses the given webhook's Config.AlternativeURLs
-// and returns as soon as the URL is considered invalid. It returns nil if the URL is
-// valid.
-func GoodAlternativeURLs(vs []ValidURLFunc) ValidatorFunc {
-	vs = filterNil(vs)
+// CheckDuration ensures that 0 <= Duration <= ttl. Duration returns an error
+// if a negative value is given.
+func CheckDuration(ttl time.Duration) ValidatorFunc {
 	return func(w Webhook) error {
-		for _, u := range w.Config.AlternativeURLs {
-			if u == "" {
-				return errInvalidAlternativeURL
-			}
-			parsedAlternativeURL, err := url.ParseRequestURI(u)
-			if err != nil {
-				return fmt.Errorf("%w: %v", errInvalidAlternativeURL, err)
-			}
-			for _, f := range vs {
-				err = f(parsedAlternativeURL)
-				if err != nil {
-					return fmt.Errorf("%w: %v", errInvalidAlternativeURL, err)
-				}
-			}
+		if ttl < 0 {
+			return errInvalidTTL
 		}
-		return nil
-	}
-}
-
-// HTTPSOnlyEndpoints creates a ValidURLFunc that considers a URL valid if the scheme
-// of the address is https.
-func HTTPSOnlyEndpoints() ValidURLFunc {
-	return func(u *url.URL) error {
-		if u.Scheme != "https" {
-			return errURLIsNotHTTPS
+		if ttl < w.Duration || w.Duration < 0 {
+			return errInvalidDuration
 		}
 		return nil
 	}
 }
 
-// RejectHosts creates a ValidURLFunc that checks the URL and ensures the
-// host does not contain any strings in the list of invalid hosts. It returns an error
-// if the host does include an invalid host name.
-func RejectHosts(invalidHosts []string) ValidURLFunc {
-	ih := []string{}
-	for _, v := range invalidHosts {
-		if v != "" {
-			ih = append(ih, v)
+// CheckUntil ensures that Until, with jitter, is not more than ttl in the future.
+func CheckUntil(jitter time.Duration, ttl time.Duration, now func() time.Time) ValidatorFunc {
+	return func(w Webhook) error {
+		if ttl < 0 {
+			return errInvalidTTL
+		} else if jitter < 0 {
+			return errInvalidJitter
 		}
-	}
-	return func(u *url.URL) error {
-		host := u.Host
-		for _, v := range ih {
-			if strings.Contains(host, v) {
-				return errInvalidHost
+		if !(w.Until).IsZero() {
```

Comment:
Rather than have all of the logic in this if statement, we can check and return when it's zero:
```go
if w.Until.IsZero() {
  // nothing to check
  return nil
}
// rest of logic here
```

Reply:
If we move the logic outside, we have to put another `return nil` line in the case that Until passes the test. Is that preferable to having the logic inside the if statement?